Objects/genericaliasobject.c

```diff
@@ -225,6 +226,33 @@ tuple_add(PyObject *self, Py_ssize_t len, PyObject *item)
     return 0;
 }
 
+// Makes a shallow copy of a tuple
+static inline PyObject *
+tuple_copy(PyObject *obj) {
+    return _PyTuple_FromArray(((PyTupleObject *)obj)->ob_item, Py_SIZE(obj));
+}
+
+// This converts all nested lists in args to a tuple (args should be a tuple).
+// Usually needed to help caching in other libraries.
+// Must be called on a copy of a tuple object, where the only reference is
+// owned by the caller.
+static inline void
+tupleify_lists(PyObject *args) {
+    Py_ssize_t len = PyTuple_GET_SIZE(args);
+    for (Py_ssize_t i = 0; i < len; ++i) {
+        PyObject *arg = PyTuple_GET_ITEM(args, i);
+        int is_list = PyList_CheckExact(arg);
+        int is_tuple = PyTuple_CheckExact(arg);
+        if (is_list || is_tuple) {
+            // In case there are lists nested inside tuples.
+            PyObject *new_arg = is_list ? PyList_AsTuple(arg) : tuple_copy(arg);
```

Comment:
The PEP 7 style guide states that lines should be limited to 79 characters.

Reply:
I'd like to argue against this, PEP 7 says 

> Note, rules are there to be broken. Two good reasons to break a particular rule:
When applying the rule would make the code less readable, even for someone who is used to reading code that follows the rules.

There are many other instances in ``genericaliasobject.c`` which exceed 79 characters (one even up to 91!).
The current line is 81 characters, I'd think a leeway of 2 characters should be afforded :), breaking it into a multi-line ternary would hurt readability IMO as it'd be better off as an if-else.

Reply:
After changing the code a bit, it no longer exceeds 79 chars, so I'm marking this as resolved :).